### DISCUSSION TITLE
Error in esphome log

### DISCUSSION THREAD

**Discussion by seaverd** [emporia-vue-local/esphome#149] [Q&A] [answered]
I am seeing the following in my esphome logs.  

[13:17:12][W][emporia_vue:067]: Detected 1 missing reading(s), data may not be accurate

Is this normal?  I don't recall seeing it when I first set this up 6 plus months ago.  I am currently having an issue where my Total Daily Energy stops for extended periods of time and I am not sure what is going on...not sure if this is related.  Seems like I have individual power values for Phase A and B and total is just the sum of those.  

 I have also attached a copy of my esphome yaml file for reference.

Any suggestions would be greatly appreciated.  

[emporiavue2_04_03_2023.txt](https://github.com/emporia-vue-local/esphome/files/11140889/emporiavue2_04_03_2023.txt)

**Comment by flaviut** [comment]
Can you include a screenshot of your data graph and attach a decent chunk of your logs (a few hundred lines)?

Occasional (every few minutes) `Detected 1 missing reading(s), data may not be accurate` is unexpected but also harmless. I think your missing data problem is unrelated.

**Reply by seaverd** [reply]
Here is the information you requested.   Let me know if you need anything else.

![emporia_vue_graphs](https://user-images.githubusercontent.com/2743685/230730784-3664cf88-cdce-4457-bec8-149a836734fb.jpg)
[ha_logs.zip](https://github.com/emporia-vue-local/esphome/files/11183624/ha_logs.zip)

**Reply by flaviut** [reply, marked as answer]
Sorry, I needed the esphome logs for the Vue, not the Home Assistant logs.

I think that `Detected 1 missing reading(s), data may not be accurate` is unrelated & harmless in this case, but there's something very wrong going on. The daily energy should be resetting every midnight, but it is not. Does https://github.com/emporia-vue-local/esphome/discussions/120#discussioncomment-5275615 solve your problem?

**Reply by seaverd** [reply]
I just changed my time settings based on #120 Comment....will see what happens.  I will also look for Esphome logs.  To also illustrate what is going on, check out these pictures from today.  Look at energy dashboard pic and you will see no energy used from say noon to 6pm.  Then look at total energy pic and you will see the flat...no change in energy value...yet the phase A and phase B show energy used during these times.   I never had this issue...not sure what update or change brought this on.


![phaseB_4_9_23](https://user-images.githubusercontent.com/2743685/230814826-7999bbb9-2506-4adf-8e9e-3d0744306b40.jpg)
![phaseA_4_9_23](https://user-images.githubusercontent.com/2743685/230814827-57990572-f344-4723-8667-08371d0f311e.jpg)
![total daily energy_4_9_23](https://user-images.githubusercontent.com/2743685/230814829-bc9a472f-a8ca-421a-b103-caa17a1263ba.JPG)
![energydashboard_4_9_23](https://user-images.githubusercontent.com/2743685/230814830-fc95db3a-8701-4bec-8bf9-066192efc148.jpg)

**Reply by seaverd** [reply]
I just wanted to confirm that the time fix you mentioned above from comment 120 worked and it reset at midnight, see attached screenshot.
![emporia_time_reset](https://user-images.githubusercontent.com/2743685/230891284-4ccd256b-577a-4c30-8f10-ac8c167480c9.jpg)

**Reply by clowrey** [reply]
I still think the example should use the home assistant time module by default ;)

**Reply by flaviut** [reply]
I'm thinking so too @clowrey. Can you make an PR if you have the time?

**Reply by seaverd** [reply]
I just want to say that after making the change to home assistant time my issue appears to be resolved.  I know its only been 30 hours or so, but it reset each night at midnight and the total energy values have made sense that entire time.

Thanks for your assistance!
Dan

**Reply by flaviut** [reply]
Glad to hear it!

I suspect the problem here is related to floating point precision. If you care for the technical details, go to http://weitz.de/ieee/, select binary32, enter in 250000000.0 ([twice your value](https://github.com/esphome/esphome/blob/80bc567c31f0b2e71e4b8750143e05d1a41de568/esphome/components/total_daily_energy/total_daily_energy.cpp)), 1.0, and press the "+" key. You get `250000000.0` out, which sort of matches the behavior here:

![image](https://user-images.githubusercontent.com/5213469/231229306-59f8789f-1899-47e3-9258-0ac7b725f46f.png)
